src/shared/conf-parser.c

```diff
@@ -1283,23 +1283,30 @@ int config_parse_strv(
                 return 0;
         }
 
+        _cleanup_strv_free_ char **strv = NULL;
         for (const char *p = rvalue;;) {
                 char *word = NULL;
 
                 r = extract_first_word(&p, &word, NULL, EXTRACT_UNQUOTE|EXTRACT_RETAIN_ESCAPE);
                 if (r == 0)
-                        return 0;
+                        break;
                 if (r == -ENOMEM)
                         return log_oom();
                 if (r < 0) {
                         log_syntax(unit, LOG_WARNING, filename, line, r, "Invalid syntax, ignoring: %s", rvalue);
                         return 0;
                 }
 
-                r = strv_consume(sv, word);
+                r = strv_consume(&strv, word);
                 if (r < 0)
                         return log_oom();
         }
+
+        r = strv_extend_strv(sv, strv, /* filter_duplicates = */ true);
```

Comment:
are you sure that in all uses of this function we actually want duplicates to be dropped? did you verify?

Reply:
maybe rename the function to config_parse_strv_uniq() or so? just to communicate that we filter dups here?

Reply:
Or, maybe, control the behavior with `ltype`.